heft-plugins/heft-lint-plugin/src/LintPlugin.ts

```diff
@@ -116,11 +134,66 @@ export default class LintPlugin implements IHeftTaskPlugin<ILintPluginOptions> {
         // Warn since don't run the linters when in watch mode.
         taskSession.logger.terminal.writeWarningLine("Linting isn't currently supported in watch mode");
       } else {
-        await Promise.all(this._lintingPromises);
+        if (!inTypescriptPhase) {
+          const fix: boolean = checkFix(taskSession, pluginOptions);
+          const sarifLogPath: string | undefined = getSarifLogPath(heftConfiguration, pluginOptions);
+          // If we are not in the typescript phase, we need to create a typescript program
+          // from the tsconfig file
+          const tsProgram: IExtendedProgram = await this._createTypescriptProgramAsync(
+            heftConfiguration,
+            taskSession
+          );
+          const rootFiles: readonly string[] = tsProgram.getRootFileNames();
+          const changedFiles: Set<IExtendedSourceFile> = new Set<IExtendedSourceFile>();
+          rootFiles.forEach((rootFilePath: string) => {
+            const sourceFile: TTypescript.SourceFile | undefined = tsProgram.getSourceFile(rootFilePath);
+            changedFiles.add(sourceFile as IExtendedSourceFile);
+          });
+
+          const lintingPromise: Promise<void> = this._lintAsync({
+            taskSession,
+            heftConfiguration,
+            fix,
+            sarifLogPath,
+            tsProgram,
+            changedFiles
+          });
+          lintingPromise.catch(() => {
+            // Suppress unhandled promise rejection error
+          });
+          // Hold on to the original promise, which will throw in the run hook if it unexpectedly fails
+          await lintingPromise;
```

Comment:
```suggestion
          await this._lintAsync({
            taskSession,
            heftConfiguration,
            fix,
            sarifLogPath,
            tsProgram,
            changedFiles
          });
```

Reply:
Done in [e59fb9a](https://github.com/microsoft/rushstack/pull/5239/commits/e59fb9a938bf50820df6b1d3ce2c3c0f60f8d030)